core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Ascii;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+
+/**
+ * A specification of a {@link Backoff} configuration represented by a string. The string syntax is
+ * a series of comma-separated options(key-values pair) and each values is colon-separated.
+ * The options are divided into two groups which are base options and decorators. Base options include
+ * {@link ExponentialBackoff}, {@link FixedBackoff} and {@link RandomBackoff}. Only one of them can be
+ * specified in the {@code specification}, otherwise {@link IllegalArgumentException} will occur.
+ * When you create a {@link Backoff} using {@link #build()}, decorators, {@link JitterAddingBackoff} and
+ * {@link AttemptLimitingBackoff}, are chaining with the base option in order even if you didn't specify
+ * them in the {@code specification}.
+ *
+ * @see Backoff#of(String) for the format of the specification, please refer to Backoff#of(String)
+ */
+final class BackoffSpec {
+    private static final Splitter OPTION_SPLITTER = Splitter.on(',').trimResults();
+    private static final Splitter KEY_VALUE_SPLITTER = Splitter.on('=').trimResults();
+    private static final Splitter VALUE_SPLITTER = Splitter.on(':').trimResults();
+
+    private static final List<String> ORDINALS = ImmutableList.of("first", "second", "third");
+
+    private static final long DEFAULT_INITIAL_DELAY_MILLIS = 200;
+    private static final long DEFAULT_MAX_DELAY_MILLIS = 10000;
+    private static final double DEFAULT_MULTIPLIER = 2.0;
+    private static final double DEFAULT_MAX_JITTER_RATE = 0.2;
+    private static final int DEFAULT_MAX_ATTEMPTS = 10;
+
+    private final String specification;
+
+    private boolean exponentialConfigured; // base option group. default base option
```

Comment:
I tried quick implementation because it sounded great. After I look at the codes, however, it just looks too much for what it worth. So, let me leave it as it is if you don't mind. :)